src/components/select/Select.js

```diff
@@ -195,7 +195,7 @@ export default class SelectComponent extends Field {
 
     if (data.data) {
       // checking additional fields in the template for the selected Entire Object option
-      const hasNestedFields = this.component.template.match(/(?<=item\.data.)\w*/g)[0];
+      const hasNestedFields = this.component.template.match(/(?<=item\.data.)\w*/g) && this.component.template.match(/(?<=item\.data.)\w*/g)[0];
```

Comment:
Why perform two regular expressions? I would prefer that you just create a variable of the result of the match, and then just use that so that it is only executed once.

Reply:
@travist, done